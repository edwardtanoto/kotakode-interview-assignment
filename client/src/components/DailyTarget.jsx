import React from 'react'

export const DailyTarget = ({ dailyTarget, setDailyTarget }) => {
    return (
        <div>
            Set your daily target
            <br />
            {/* No negative validation - DONE */}
            <input value={dailyTarget} min={1} onKeyDown={(e) => e.preventDefault()} style={{ width: "200px", borderRadius: '10px', background: 'transparent' }} className="daily-form pl-3" type="number" placeholder="5" onChange={(e) => setDailyTarget(e.target.value)} />

        </div>
    )
}
