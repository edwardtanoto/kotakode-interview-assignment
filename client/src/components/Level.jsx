import React from 'react'

export const Level = ({ level, role }) => {
    return (
        <div>
            Level: {`${level}`} <br />
            Role: {`${role[level]}`}
        </div>
    )
}
