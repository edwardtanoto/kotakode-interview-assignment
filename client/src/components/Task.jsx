import React from 'react'

export const Task = ({ task, deleteTask, index, completeTask }) => {

    return (
        <>
            {
                task.completed == false ?
                    <>
                        <li className="list-group-item mt-3 mb-1 pl-3">
                            {task.title}
                            <span
                                onClick={() => deleteTask(task.id)} className="btn btn-danger btn-sm float-right">x
                            </span>
                            <span onClick={() => completeTask(index)} className="btn btn-success btn-sm float-right">&#10003;
                            </span>
                        </li >
                    </> :
                    <>
                        <li className="list-group-item mt-3 mb-1 pl-3" style={{ backgroundColor: "#FFCECE" }}>
                            {task.title}
                            <span
                                onClick={() => deleteTask(task.id)} className="btn btn-danger btn-sm float-right">x
                            </span>
                        </li >
                    </>
            }

        </>
    )
}
