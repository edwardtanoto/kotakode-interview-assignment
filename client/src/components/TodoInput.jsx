import React, { useState, useContext } from 'react'
import { GlobalContext } from '../context/GlobalState';

export const TodoInput = () => {
    const [title, setTitle] = useState('');
    const id = Math.floor(Math.random() * Date.now())
    let completed = false

    const { addTask } = useContext(GlobalContext);

    const onSubmit = e => {
        e.preventDefault();

        if (title.trim()) {
            const newTask = {
                title,
                id,
                completed
            };
            addTask(newTask)
            setTitle('')
        } else {
            window.alert('You have not add any item.')
            setTitle('')
        }
    }

    return (
        <form onSubmit={onSubmit} className="mt-5">
            <div className="d-flex justify-content-center">
                <input className="form-control" style={{ width: "400px" }} type="text" value={title} onChange={(e) => setTitle(e.target.value)} placeholder="Enter text..." />
                <button className="btn btn-primary btn-small" style={{ borderRadius: '0px' }} type="submit">+</button>
            </div>
        </form >
    )
}

