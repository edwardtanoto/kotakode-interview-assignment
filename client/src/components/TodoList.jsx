import React from 'react'
import { Task } from './Task'



export const TodoList = ({ tasks, deleteTask, completeTask }) => {
    return (
        <div className="mt-5">
            <h5>To Do</h5>
            <ul className="list-group list-group-flush">
                {
                    // repeating my codes here
                    tasks.sort((a, b) => {
                        if (a.completed || !b.completed) {
                            return 1;
                        }
                        else if (b.completeTask || !a.completed) return -1
                    }).map((task, index) => <Task key={task.id} task={task} index={index} deleteTask={deleteTask} completeTask={completeTask} />)
                }
            </ul>
        </div>
    )
}
