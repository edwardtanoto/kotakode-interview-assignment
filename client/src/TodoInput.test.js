import React from "react";
import renderer from "react-test-renderer";
import { TodoInput } from "./components/TodoInput";

describe("TodoInput", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<TodoInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
