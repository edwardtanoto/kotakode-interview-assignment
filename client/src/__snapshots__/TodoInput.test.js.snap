// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput renders correctly 1`] = `
<form
  className="mt-5"
  onSubmit={[Function]}
>
  <div
    className="d-flex justify-content-center"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="Add Task"
      style={
        Object {
          "width": "400px",
        }
      }
      type="text"
      value=""
    />
    <button
      className="btn btn-primary btn-small"
      style={
        Object {
          "borderRadius": "0px",
        }
      }
      type="submit"
    >
      +
    </button>
  </div>
</form>
`;
